packages/sdk/src/swap/sdk.ts

```diff
@@ -511,4 +524,55 @@ export class SwapSDK {
       fillOrKillParams: inputFoKParams,
     };
   }
+
+  async getVaultSwapData({
+    quote,
+    srcAddress,
+    destAddress,
+    fillOrKillParams: inputFoKParams,
+    affiliateBrokers: affiliates,
+    ccmParams,
+    brokerCommissionBps,
+    extraParams,
+  }: VaultSwapRequest) {
+    await this.validateSwapAmount(quote.srcAsset, BigInt(quote.depositAmount));
+    assertQuoteValid(quote);
+
+    const vaultSwapRequest = {
+      srcAsset: quote.srcAsset,
+      destAsset: quote.destAsset,
+      srcAddress,
+      destAddress,
+      amount: quote.depositAmount,
+      ccmParams,
+      maxBoostFeeBps: 'maxBoostFeeBps' in quote ? quote.maxBoostFeeBps : undefined,
+      fillOrKillParams: parseFoKParams(inputFoKParams, quote)!,
+      dcaParams: quote.type === 'DCA' ? quote.dcaParams : undefined,
+      extraParams,
+    };
+
+    if (this.options.broker) {
+      return requestSwapParameterEncoding(
+        {
+          ...vaultSwapRequest,
+          commissionBps: brokerCommissionBps ?? this.options.broker.commissionBps,
+          affiliates,
+        },
+        { url: this.options.broker.url },
+        this.options.network,
+      );
+    }
+
+    assert(vaultSwapRequest.fillOrKillParams, 'fill or kill params are required');
+    assert(
+      !affiliates?.length,
+      'Broker commission is only supported only when initializing the SDK with a brokerUrl',
+    );
+    assert(
```

Comment:
will add an option to get vault swap data with a commission and a broker id in a follow up